### ISSUE TITLE
Spurious characters in OpenAPI / Swagger UI parameter names

### ISSUE THREAD

**Issue by inad9300** [OpenLiberty/open-liberty#26775]
**Describe the bug**  
Unexpected "Â" characters appear next to every parameter name of the generated OpenAPI UI (while not in the YAML):

![image](https://github.com/OpenLiberty/open-liberty/assets/3440331/d23ef4e1-55fd-4f21-977b-f7e0fd4a2409)

Anecdotally, there are some other spurious "Â"s throughout your codebase, although none show up related to Swagger: https://github.com/search?q=repo%3AOpenLiberty%2Fopen-liberty%20%C3%82&type=code, e.g.

![image](https://github.com/OpenLiberty/open-liberty/assets/3440331/7b16fe33-04eb-4aba-80d4-2d3a3b16e4a2)

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.10
 - Affected feature(s): mpOpenAPI-3.1
 - Java Version: 17.0.8

**Comment by Azquelt** [comment]
Thank you, we will take a look.

**Comment by fmhwong** [comment]
Similar to https://github.com/OpenLiberty/open-liberty/issues/26655

**Comment by abutch3r** [comment]
Reviewing the JS in the browser debugger for required parameters we have this segment of the javascript:
```
("div",{className:W?"parameter__name required":"parameter__name"},r.get("name"),W?v.default.createElement("span",null,"Â *"):null)
```

We should be getting `&nbsp;` in place of the spurious character, it is known that `&nbsp;` and `Â` can occur depending on the encoding in question. When looking at the actual bundle in then we can see a different form of `NBSP` in various tooling. so falls under an encoding issue when the js file is built as the swagger-ui source has the correct sequence

**Comment by abutch3r** [comment]
The issue is that the JS is being interpreted as ISO-8859-1 instead of UTF-8 when retrieved by the browser. No request or response headers have changed and the recent JS changes are in previous releases and do not exhibit the issue. adding `charset="UTF-8"` into the index html files forces the browsers to correctly interpret the javascript and no longer get the spurious character.

Given that also we set <meta charset="UTF-8">. It should not be needed to additionally set the `meta` tag, but this is what Swagger-UI do for the example petstore with the meta tag and include the charset on the script tag

Checked for Firefox 118 and Chrome 118

**Comment by abutch3r** [comment]
The Issue has been found to start occurring in 23.0.0.9. It has been found that when the request is made for `/openapi/ui` the `content-type` header was now including `charset=ISO-8859-1` which the `<meta>` tag was unable to override and all subsequent files being requested were interpreted using the ISO charset instead of UTF-8.

The most likely cause is the new Servlet filter which does some overriding of the html before it reaches the Server. As is is a Servlet component the default is ISO-8859-1, so via the manipulation it was doing it is likely it was adding the charset to the header. There was a fix in 23.0.0.9 to set the content-length of the response in the filter which is likely the cause of the change in behavior.
